### PR TITLE
Add array as a valid parameter for flashbag add() method.

### PR DESCRIPTION
**Pull request by ncareau**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

Adding `array` as a valid parameter for message in flashbag. It was already added to FlashbagInterface.set() but was missing for the add method. Also corrected in ControllerTrait.

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/Session/Flash/FlashBagInterface.php [hunk 1]
```diff
@@ -24,7 +24,7 @@
      * Adds a flash message for type.
      *
      * @param string $type
-     * @param string $message
+     * @param string|array $message
```
Comment (ro0NL) [inline src/Symfony/Component/HttpFoundation/Session/Flash/FlashBagInterface.php]: Looking at https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/Session/Flash/FlashBag.php#L73 i dont think this is supported =/

**Comment by xabbuh** [comment]
I am 👎 on this change as changing the argument type in the interface would be a BC break (all implementing class would be updated to support array parameters).

**Comment by ncareau** [comment]
My premise was that i can put arrays has message and get multiple parameter for a single message.

For example :
`$this->addFlash('message', [
'type' => 'error', 
'icon' => 'exclamation',
'content' => $this->get('translator')->trans('An error has occured')
]);`

This works now, but i get validation error since it's expecting a string. 

Also seen for example in : 
https://stackoverflow.com/a/36496077/1207809

**Comment by HeahDude** [comment]
Agreed with @xabbuh, not sure this is with it. And this should be pretty easy to handle such methods in a custom controller trait in user land.

**Comment by javiereguiluz** [comment]
I'm 👎  about this proposal too. @ncareau let me comment you two things:

1) If `set()` supports arrays and `add()` doesn't is because according to [Symfony naming conventions](https://symfony.com/doc/current/contributing/code/conventions.html) and the practices we follow in our code, `add()` simply adds 1 element to a collection and `set()` replaces the entire collection.

2) This is your original code and why you needed an array instead of a string:

```php
$this->addFlash('message', [
    'type' => 'error',
    'icon' => 'exclamation',
    'content' => $this->get('translator')->trans('An error has occurred')
]);
```

Luckily, there's an even simpler solution to achieve that and not use an array. It's also the practice we recommend in Symfony Docs:

```php
$this->addFlash('error', $this->get('translator')->trans('An error has occurred'));
```

The `error` message type is commonly used as the CSS class name, so you can customize the design, the icon, etc. I hope this idea helps you solve this issue.

**Pull request closed without merging**